Mathlib/FieldTheory/Separable.lean

```diff
@@ -625,58 +628,105 @@ instance Algebra.IsSeparable.isAlgebraic [Nontrivial F] [Algebra.IsSeparable F K
 
 end CommRing
 
-instance Algebra.isSeparable_self (F : Type*) [Field F] : Algebra.IsSeparable F F :=
-  ⟨fun x => by
-    rw [IsSeparable, minpoly.eq_X_sub_C']
-    exact separable_X_sub_C⟩
+section
+
+variable {F K : Type*} [Field F] [Ring K] [Algebra F K]
+
+theorem isSeparable_algebraMap {x : F} : IsSeparable F (algebraMap F K x) :=
+  Polynomial.Separable.of_dvd (Polynomial.separable_X_sub_C (x := x))
+    (minpoly.dvd F (algebraMap F K x) (by simp only [map_sub, aeval_X, aeval_C, sub_self]))
+
+variable (F) in
+instance Algebra.isSeparable_self : Algebra.IsSeparable F F :=
+  ⟨fun _ => isSeparable_algebraMap⟩
 #align is_separable_self Algebra.isSeparable_self
 
+end
+
+section
+
+variable (F K : Type*) [Field F] [Field K]
+    [Algebra F K] [FiniteDimensional F K] [CharZero F]
+
+variable {K} in
+theorem IsSeparable.of_finite (x : K) : IsSeparable F x :=
+  (minpoly.irreducible <| .of_finite F x).separable
+
 -- See note [lower instance priority]
 /-- A finite field extension in characteristic 0 is separable. -/
-instance (priority := 100) Algebra.IsSeparable.of_finite (F K : Type*) [Field F] [Field K]
+protected instance (priority := 100) Algebra.IsSeparable.of_finite (F K : Type*) [Field F] [Field K]
     [Algebra F K] [FiniteDimensional F K] [CharZero F] : Algebra.IsSeparable F K :=
-  ⟨fun x => (minpoly.irreducible <| .of_finite F x).separable⟩
+  ⟨fun _ => _root_.IsSeparable.of_finite _ _⟩
 #align is_separable.of_finite Algebra.IsSeparable.of_finite
 
+end
+
 section IsSeparableTower
 
-/-- If `R / K / A` is an extension tower, `x : R` is separable over `A`, then it's also separable
+section
+
+variable (F K E : Type*) [CommRing F] [Field K] [CommRing E] [Algebra F K]
+    [Algebra F E] [Algebra K E] [IsScalarTower F K E]
+
+variable {F E} in
+/-- If `E / K / F` is an extension tower, `x : E` is separable over `F`, then it's also separable
 over `K`. -/
-theorem IsSeparable.of_isScalarTower {A : Type*} [CommRing A]
-    (K : Type*) [Field K] [Algebra A K] {R : Type*} [CommRing R] [Algebra A R] [Algebra K R]
-    [IsScalarTower A K R] {x : R} (h : IsSeparable A x) : IsSeparable K x :=
+theorem IsSeparable.tower_top
+    {x : E} (h : IsSeparable F x) : IsSeparable K x :=
```

Comment:
Was this rename discussed somewhere? It probably deserves to have deprecated aliases for the old names.

Reply:
The general rename issue from algebra to element is discussed at [here](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/.60IsSeparable.60.20and.20.60Algebra.2EIsSeparable.60). But there is no specific discussion for this theorem.

Reply:
And yes, I believe creating deprecated aliases for the old names is a good idea. Do you think we should do deprecated aliases for all renamed theorems? And for the previous PR #14404 ?

Reply:
I've checked all theorems changes in this PR. Only `IsSeparable.of_isScalarTower` and `AlgEquiv.isSeparable` are possible to have a deprecated alias. Other theorem's old names are occupied.